src/averageAges.js

```diff
@@ -15,11 +15,27 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => person.sex === 'm');
+  let result = 0;
+
+  if (!century) {
+    const ageOfDeath = men.map(person => person.died - person.born);
+    const sumAge = ageOfDeath.reduce((sum, curent) => sum + curent, 0);
+
+    result = sumAge / men.length;
+  } else {
+    const startCenture = century - 1 + '01';
+    const endCenture = century + '00';
+    const centureResultArr = men.filter(pers =>
+      pers.died > startCenture && pers.died < endCenture
```

Comment:
no need to find start and end century, you can use `Math.ceil(person.died / 100)` to find the person's century of death and then `Math.ceil(person.died / 100) === century` to compare them